### PR TITLE
Rewrite listener volume docs

### LINKED ISSUE #496: Clarify documentation for the listener volume scope
https://github.com/stackabletech/operator-rs/pull/858#issuecomment-2333394725

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #496 (hopefully).

Also moves the listener volume scope to the top of the list and adds caveats to the other scopes.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by nightkr** [comment]
Particularly curious to hear @razvan's thoughts, though I think it's still a bit more abstract than would be ideal...

File: docs/modules/secret-operator/pages/scope.adoc [hunk 1]
```diff
@@ -12,34 +12,46 @@ The exact effect of the scope depends on which xref:secretclass.adoc#backend[`ba
...
+volume as its parameter. This way, the scope will automatically adapt to the xref:listener-operator:listenerclass.adoc[ListenerClass] that it uses.
+
+Multiple xref:listener-operator:listener.adoc[Listener]s can be bound by specifying the scope multiple times.
+
+note:: The parameter is the name of the Pod's _volume_, not the name of the PersistentVolumeClaim, PersistentVolume, or xref:listener-operator:listener.adoc[Listener]. This allows the scope to bind to replica-specific or ephemeral xref:listener-operator:listener.adoc[Listener]s.
 
 [#pod]
 === `pod`
 
+*Syntax*: `node`
```
Comment (soenkeliebau) [inline docs/modules/secret-operator/pages/scope.adoc:32]: typo?

Comment (nightkr) [reply docs/modules/secret-operator/pages/scope.adoc]: Goddamnit, copy/paste is dangerous
f17cea74efa9b9c120fec8e838588a2ada62336e

**Review by razvan — APPROVED** [review]
thanks!

**Pull request merged**